docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/03_chart_html_api.mdx

```diff
@@ -0,0 +1,430 @@
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>element:click</td>
+      <td><code>Custom Event</code></td>
+      <td>Adds click events to the element of the chart.</td>
+<td>
+
+```typescript
+<rapid-g2plot-chart @element:click="${(x) => x.onElementClick()}">
+```
+
+</td>
+    </tr>
+    <tr>
+      <td>legend-item:click</td>
+      <td><code>Custom Event</code></td>
+      <td>Adds click events to the legened element.</td>
+<td>
+
+```typescript
+<rapid-g2plot-chart @legend-item:click="${(x) => x.onLegendClick()}">
+```
+
+</td>
+    </tr>
+  </tbody>
+</table>
+
+For a complete list of available events, please visit the [G2Plot Events Documentation](https://g2plot.antv.antgroup.com/en/api/options/events).
+
+### Listened Events
+
+This component doesn't listen to any public API events.
+
+## Chart Datasource
+
+The `chart-datasource` element is used to declaratively set the data source for a `rapid-g2plot-chart`.
+
+```html
+<rapid-g2plot-chart>
+  <chart-datasource></chart-datasource>
+</rapid-g2plot-chart>
+```
+
+### DOM API
+
+Property and attribute binding examples for Genesis Component syntax. Closing tag omitted.
+
+#### Attributes
+
+<table>
+    <thead>
+        <tr>
+            <th>Attribute</th>
+            <th>Type</th>
+            <th>Use</th>
+            <th>Example</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>chart-fields</td>
+            <td><code>string</code></td>
+            <td>Defines the fields for chart configuration.<ol><li>First is always the xField/colorField (**Mandatory**)</li><li>Second is always the yField/angleField (**Mandatory**)</li><li>Third is always the seriesField (**Optional**)</li></ol>Default value is 'groupBy value'. Can be customized but order matters.</td>
+<td>
+
+```typescript
+<chart-datasource chart-fields="groupBy value">
+```
+
+</td>
+        </tr>
+        <tr>
+            <td>server-fields</td>
+            <td><code>string</code></td>
+            <td>Defines the fields of data to be charted.<ol><li>First will be associated to the first value in 'chartFields' (xField/colorField) (**Mandatory**)</li><li>Second will be associated to the second value in 'chartFields' (yField/angleField) (**Mandatory**)</li><li>Third will be associated to the third value in 'chartFields' (**Optional**)</li></ol>Can be customized but order matters.</td>
+<td>
+
+```typescript
+<chart-datasource server-fields="TRADE_ID QUANTITY">
+```
+
+</td>
+        </tr>
+        <tr>
+            <td>resourceName</td>
+            <td><code>string</code></td>
+            <td>Name of the backend resource which contain the data.</td>
+<td>
+
+```typescript
+<chart-datasource resourceName="ALL_TRADES">
+```
+
+</td>
+        </tr>
+        <tr>
+            <td>criteria</td>
+            <td><code>string</code></td>
+            <td>Clients can send a Groovy expression to perform filters on the query server, these remain active for the life of the subscription.</td>
```

Comment:
Is there a link to any server pages which cover writing these filters in more detail we could link to here?

Reply:
@SzymonZur I have reached out to Patrick about this on Slack and he's going to create a reuseable section on this we can include in appropriate pages